README.md

```diff
@@ -158,21 +171,73 @@ class MyView @JvmOverloads constructor(
 
     init {
         if (!isInEditMode) {
-            Whetstone.inject(view = this)
+            Whetstone.inject(this)
         }
     }
 }
 ```
 
-### Worker / WorkManager (DONE)
-To use Whetstone's work manager integration, a separate dependency is required:
+### Fragments
+Fragments support only construction injection, exclusively. This is possible because we are able to hook into the system to influence exactly how fragments should be created. To achieve this, the activity hosting the fragment must install Whetstone's fragment factory.
 
 ```kotlin
-implementation("com.deliveryhero.whetstone:whetstone-worker")
+class MyActivity : Activity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        Whetstone.inject(this)
+        super.onCreate(savedInstanceState)
+    }
+}
+```
+
+Then you're able to use the injected fragments with as many constructor arguments as necessary, as long as all these dependencies can be satisfied by dependency injection.
+
+```kotlin
+@ContributesFragment
+class MyFragment @Inject constructor(
+    private val dependency: MyDependency,
+    private val anotherDependency: AnotherDependency,
+): Fragment() {
+
+    // Get the contributed ViewModel
+    // We automatically handle process death and saved state handle wiring
+    private val viewModel by injectedViewModel<MyViewModel>()
+    private val activityViewModel by injectedActivityViewModel<ActivityViewModel>()
+}
+```
+
+Note that all injected fragments must be created via the fragment manager. For example:
+
+```kotlin
+val myFragment = fragmentManager.instantiate<MyFragment>()
 ```
 
-This will automatically install Whetstone's worker factory (replacing the default factory), so that you can immediately
-start taking advantage of injected workers
+For fragments that don't require any external dependencies, the simple no-arg constructor can still be used, and we gracefully fallback to the default behavior
+
+**Important:** A Fragment should **NEVER** be scoped. The Android Framework controls the Lifecycle of **ALL** Fragments.
+
+### ViewModels
+Like fragments, view models also support full constructor injection
+
+```kotlin
+@ContributesViewModel
+class MyViewModel @Inject constructor(
+    private val savedStateHandle: SavedStateHandle,
+): ViewModel()
+```
+
+**Important:** A ViewModel should **NEVER** be scoped. The Android Framework controls the Lifecycle of **ALL** ViewModels.
+
+### WorkManager
+Workmanager integration is an extra add-on, and must be enabled explicitly in your `build.gradle` file before use:
+
+```kotlin
+whetstone {
+    addOns.workManager.set(true)
+}
```

Comment:
(nitpick) This looks nice, although the camel case for `addOns` looks weird 😅 I think I've seen it used as `addons` even though not technically correct would be okay to use as configuration/input

(I know I should've reviewed the plugin PR)

Reply:
Hmm. If you're focusing on the `addOns` in particular, I believe it's typically written as `add-on`, hence why I used `addOns`
But if you're thinking of having configs all lower case? Something like: `addons.workmanager.set(true)`, I'm not sure that I particularly like it (personal opinion). I'll be open to a different name altogether though, something other than addOns 🤔